src/waitress/wasyncore.py

```diff
@@ -151,31 +151,45 @@ def poll(timeout=0.0, map=None):
     if map is None:  # pragma: no cover
         map = socket_map
     if map:
-        r = []
-        w = []
-        e = []
-        for fd, obj in list(map.items()):  # list() call FBO py3
-            is_r = obj.readable()
-            is_w = obj.writable()
-            if is_r:
-                r.append(fd)
-            # accepting sockets should not be writable
-            if is_w and not obj.accepting:
-                w.append(fd)
-            if is_r or is_w:
-                e.append(fd)
-        if [] == r == w == e:
-            time.sleep(timeout)
-            return
+        fds = list(map.keys())
```

Comment:
`map` should be a dictionary, so this should return the list of keys:
```suggestion
        fds = list(map)
```